From 4006d530feb04d3d23d87de3b216ec520cc189dc Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Fri, 5 Feb 2010 17:31:30 -0200
Subject: [PATCH 14/14] vnc: Use inet_strfamily()

RH-Author: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: <1265391090-21418-4-git-send-email-lcapitulino@redhat.com>
Patchwork-id: 6937
O-Subject: [PATCH 3/3] vnc: Use inet_strfamily()
Bugzilla: 562181
RH-Acked-by: Juan Quintela <quintela@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
(cherry picked from commit dc0d4efc02188ad9e5785adaba874bfd198eee76)
---
 vnc.c |   22 +---------------------
 1 files changed, 1 insertions(+), 21 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 vnc.c |   22 +---------------------
 1 files changed, 1 insertions(+), 21 deletions(-)

diff --git a/vnc.c b/vnc.c
index a042f24..2cc4b7c 100644
--- a/vnc.c
+++ b/vnc.c
@@ -100,26 +100,6 @@ char *vnc_socket_remote_addr(const char *format, int fd) {
     return addr_to_string(format, &sa, salen);
 }
 
-static QString *get_sock_family(const struct sockaddr_storage *sa)
-{
-    const char *name;
-
-    switch (sa->ss_family)
-    {
-        case AF_INET:
-            name = "ipv4";
-            break;
-        case AF_INET6:
-            name = "ipv6";
-            break;
-        default:
-            name = "unknown";
-            break;
-    }
-
-    return qstring_from_str(name);
-}
-
 static int put_addr_qdict(QDict *qdict, struct sockaddr_storage *sa,
                           socklen_t salen)
 {
@@ -138,7 +118,7 @@ static int put_addr_qdict(QDict *qdict, struct sockaddr_storage *sa,
 
     qdict_put(qdict, "host", qstring_from_str(host));
     qdict_put(qdict, "service", qstring_from_str(serv));
-    qdict_put(qdict, "family", get_sock_family(sa));
+    qdict_put(qdict, "family",qstring_from_str(inet_strfamily(sa->ss_family)));
 
     return 0;
 }
-- 
1.6.3.rc4.29.g8146

